frontend/src/concepts/roleBinding/RoleBindingPermissionsTable.tsx

```diff
@@ -40,12 +42,31 @@ const RoleBindingPermissionsTable: React.FC<RoleBindingPermissionsTableProps> =
   permissions,
   permissionOptions,
   typeAhead,
+  isProjectSubject,
   isAdding,
   onDismissNewRow,
   onError,
   refresh,
 }) => {
+  const { projects } = React.useContext(ProjectsContext);
   const [editCell, setEditCell] = React.useState<string[]>([]);
+
+  const getSubjectDisplayName = React.useCallback(
+    (name: string): string =>
+      projects.find((project) => project.metadata.name === name)?.metadata.annotations?.[
+        'openshift.io/display-name'
+      ] || name,
+    [projects],
+  );
+
+  const getSubjectResourceName = React.useCallback(
+    (name: string): string =>
+      projects.find(
+        (project) => project.metadata.annotations?.['openshift.io/display-name'] === name,
+      )?.metadata.name || name,
+    [projects],
+  );
+
```

Comment:
Sorry to give you one more round of changes --

`getSubjectDisplayName` and `getSubjectResourceName` only apply to projects, and every usage of them is behind `isProjectSubject &&`. I think now that we're defining them in here instead of lifting them out, we're losing the benefit of the abstraction and we have a lot of unnecessary prop-drilling. We already have `isProjectSubject` drilled down into each of the places we're using these, and those components could just consume `ProjectsContext` directly.

I think maybe what we should do is change these to static utils - maybe add them to `frontend/src/concepts/projects/utils.ts`, make them take the projects array and name them more explicitly. You can also use `getDisplayNameFromK8sResource` from `~/concepts/k8s/utils` for the first one.

Maybe something like this:

```ts
export const namespaceToProjectDisplayName = (
  namespace: string,
  projects: ProjectKind[],
): string => {
  const project = projects.find((p) => p.metadata.name === namespace);
  return project ? getDisplayNameFromK8sResource(project) : namespace;
};

export const projectDisplayNameToNamespace = (
  displayName: string,
  projects: ProjectKind[],
): string => {
  const project = projects.find(
    (p) => p.metadata.annotations?.['openshift.io/display-name'] === displayName,
  );
  return project?.metadata.name || displayName;
};
```

And then you could get rid of all the props where you pass around `getSubjectDisplayName` and `getSubjectResourceName`, and instead consume ProjectsContext as close to where you call these as possible so you have the projects array to pass into them. I think that would be less prop drilling?

---

Now that I write all that, I'm questioning it. If you disagree with the value of that refactor, can you instead just keep what you have but rename `getSubjectDisplayName` to `getProjectDisplayName` and `getSubjectResourceName` to `getProjectResourceName`? I think that would at least solve the readability.

I'm fine with either solution, I leave it up to you. Thanks for bearing with my rambling.

Reply:
Yeah, this makes more sense. I have updated this. Thanks!!